qiskit_optimization/converters/indicator_to_inequality.py

```diff
@@ -0,0 +1,179 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""The inequality to equality converter."""
+
+import copy
+import logging
+from typing import List, Optional, Union
+
+import numpy as np
+
+from .quadratic_program_converter import QuadraticProgramConverter
+from ..exceptions import QiskitOptimizationError
+from ..problems.constraint import Constraint
+from ..problems.quadratic_objective import QuadraticObjective
+from ..problems.quadratic_program import QuadraticProgram
+from ..problems.variable import Variable
+
+logger = logging.getLogger(__name__)
+
+
+class IndicatorToInequality(QuadraticProgramConverter):
+    """Convert inequality constraints into equality constraints by introducing slack variables.
+
+    Examples:
+        >>> from qiskit_optimization.problems import QuadraticProgram
+        >>> from qiskit_optimization.converters import IndicatorToInequality
+        >>> problem = QuadraticProgram()
+        >>> # define a problem
+        >>> conv = IndicatorToInequality()
+        >>> problem2 = conv.convert(problem)
+    """
+
+    _delimiter = "@"  # users are supposed not to use this character in variable names
+
+    def __init__(self) -> None:
+        self._src = None  # type: Optional[QuadraticProgram]
+        self._dst = None  # type: Optional[QuadraticProgram]
+
+    def convert(self, problem: QuadraticProgram) -> QuadraticProgram:
+        """Convert a problem with indicator constraints into one with only inequality constraints.
+
+        Args:
+            problem: The problem to be solved, that may contain indicator constraints.
+
+        Returns:
+            The converted problem, that contain only inequality constraints.
+
+        Raises:
+            QiskitOptimizationError: If a variable type is not supported.
+            QiskitOptimizationError: If an unsupported mode is selected.
+            QiskitOptimizationError: If an unsupported sense is specified.
+        """
+        self._src = copy.deepcopy(problem)
```

Comment:
Just a question. Why is this `deepcopy` needed?

Reply:
It used to be necessary. Matsuo-san can remove it.